### ISSUE TITLE
Photos result is not as shown in README

### ISSUE THREAD

**Issue by orasik** [NVIDIA/FastPhotoStyle#26]
Hi,
First of all thank you very much for sharing the code and for good documentation. I was able to run the dockerised app easily on Ubuntu 17 without any issues. 

Regarding results, photos are not as shown in your README page. I got totally different results. Please check the following examples below:

**Content:**
![forest-small](https://user-images.githubusercontent.com/9113844/36875352-8153195e-1da7-11e8-9596-5b7c6e493ca1.png)

**Style:**
![redforest-small](https://user-images.githubusercontent.com/9113844/36875343-79b160a2-1da7-11e8-93c9-05f016ae1418.png)

**Output:**

![forest](https://user-images.githubusercontent.com/9113844/36875330-6e3fa2b0-1da7-11e8-8a1e-1c2ad2731f0c.png)

Please note that I haven't used the labelme, just the general one.

Is that expected result? or there is something I am missing?

Kind Regards,
Oras

**Comment by Yijunmaverick** [comment]
@orasik Thanks for your interests. As stated in the paper, we use label maps for most pairs, as did in the [DPST](https://github.com/luanfujun/deep-photo-styletransfer) work. I draw a simple 2-label (ground and non-ground) map for you and the result looks like this:

![in130](https://user-images.githubusercontent.com/10704714/36877144-91c7c3be-1d6d-11e8-88d3-98efcc680c7f.png)

Hope you like it.

**Comment by orasik** [comment]
Fantastic thanks @Yijunmaverick 
Would you mind sharing the json files for labels?

Kind Regards,
Oras

**Comment by chiachunfu** [comment]
I created my own label.png (3-label) without labelme and ran into the following error when using demo.py with label maps:
  File "demo.py", line 43, in <module>
    cuda=args.cuda,
  File "/home/jjfu/code/FastPhotoStyle/process_stylization.py", line 62, in stylization
    stylized_img = p_wct.transform(cont_img, styl_img, cont_seg, styl_seg)
  File "/home/jjfu/code/FastPhotoStyle/photo_wct.py", line 35, in transform
    csF4 = self.__feature_wct(cF4, sF4, cont_seg, styl_seg)
  File "/home/jjfu/code/FastPhotoStyle/photo_wct.py", line 82, in __feature_wct
    t_cont_seg = np.asarray(Image.fromarray(cont_seg, mode='RGB').resize((cont_w, cont_h), Image.NEAREST))
  File "/opt/anaconda2/lib/python2.7/site-packages/PIL/Image.py", line 2380, in fromarray
    return frombuffer(mode, size, obj, "raw", rawmode, 0, 1)
  File "/opt/anaconda2/lib/python2.7/site-packages/PIL/Image.py", line 2333, in frombuffer
    return frombytes(mode, size, data, decoder_name, args)
  File "/opt/anaconda2/lib/python2.7/site-packages/PIL/Image.py", line 2266, in frombytes
    im.frombytes(data, decoder_name, args)
  File "/opt/anaconda2/lib/python2.7/site-packages/PIL/Image.py", line 758, in frombytes
    raise ValueError("not enough image data")

Perhaps there is something I did wrong when generating the label map? Any advice would be appreciated. Thanks!

**Comment by AliFlux** [comment]
![download](https://user-images.githubusercontent.com/8333794/37282035-d65f9e34-2614-11e8-9618-b4865b6de2fd.jpg)

Here is my result, it's a bit disappointing since these input and style image are identical in their composition. Is this the expected output for an unlabeled set?

**Comment by mingyuliutw** [comment]
It looks like the black regions in the top and bottom of the style image in your display is part of the style image. This could be the issue.

**Comment by Yijunmaverick** [comment]
@AliFlux Right, the black frame is also counted as part of the style. I remove it and give a trial:

No label map:
![in141](https://user-images.githubusercontent.com/10704714/37313740-d74c6400-260d-11e8-925c-292d0f73bf3a.png)

With a 3-label map (sky, sea, and rest):
![in140](https://user-images.githubusercontent.com/10704714/37313762-e46dcc00-260d-11e8-99c4-7a6d9248d755.png)

Thanks.

**Comment by AliFlux** [comment]
Yeah, my mistake. The style image had black border and it was causing issues. Also, is there a way to minimize or avoid the smudges/spots on the sky and the land? Almost none of the images in README.md have such artifacts.

![Demo Images](https://github.com/NVIDIA/FastPhotoStyle/raw/master/teaser.png)

Thanks.

**Comment by mingyuliutw** [comment]
I think the first stylization result in the figure also has the "spot artifacts." We are also seeking ways to improve our method. If you figure out how to minimize these artifacts, please share with us.

**Comment by asdfqwer2015** [comment]
Hi, I tested these images from Adobe's Deep Photo Style Transfer's readme, but the result seems not very well . Could you please check the reason why these case have not enough style transfer? thx.
![case1](https://user-images.githubusercontent.com/26498687/38013053-51636efc-3296-11e8-9434-d8164310c357.jpg)
![case3](https://user-images.githubusercontent.com/26498687/38014719-32098504-329c-11e8-9ef7-19dc52ae52cf.jpg)
In these two case, Fast style transfer's result seems without enough style transfer. Why they haven't enough style transfer? And I tested more face style transfer case, and they seems haven't enough style transfer too.


![case2](https://user-images.githubusercontent.com/26498687/38014333-f315e0dc-329a-11e8-816d-ee0ee6699049.jpg)
And In this case, Why transfer with segs is worse than without segs?

**Comment by wesleyw72** [comment]
This may not help, but certainly worth trying. Tweak the beta value found on line 16 of photo_smooth.py. It balances between the smoothness and the fitting term of the photorealistic smoothing step. It might be best to read the author's to see how it works, in particular look at equation 4 and the section "Sensitivity analysis on λ".

In short, beta=1/(1+λ). So try beta=0.999 or 0.99 and see what happens.

**Comment by Yijunmaverick** [comment]
@asdfqwer2015 Thanks for your interests. In addition to suggestions from @wesleyw72, I would like to explain more:

Case 1: First, when you show the result of DPST here for comparison, make sure you are using the same mask as did in DPST. Directly comparing the result between w/ seg and w/o seg is somehow unfair. Second, we always need to find a balance between style transferring and content preserving. Therefore, sometimes detailed textures (e.g., freckle) in the style cannot be transferred well. You can adjust the parameters, as indicated by @wesleyw72, to find your favorite point.

For your reference, there is a great [paper](http://people.csail.mit.edu/yichangshih/portrait_web/) that specifically works on portrait style transfer. Our algorithm is more towards general scenery and there is still room for further improvements.

Case 2: When you show the result of DPST here for comparison, make sure you are using the same mask as did in DPST. Check their style [seg](https://github.com/luanfujun/deep-photo-styletransfer/blob/master/examples/segmentation/tar58.png) on this example. Here is our result using the same mask:

![in58](https://user-images.githubusercontent.com/10704714/38045265-27761358-3271-11e8-9e81-53e0a9e9d69a.png)

Case 3: I am a little bit confused. With the mask, I guess you want the black bed to be exactly stylized by the yellow bed. In terms of style transferring, I think the result on the bottom row reflect your intention well.

**Comment by asdfqwer2015** [comment]
Thx for the quick and detailed reply.
I've tried different λ. 

And thx for your recommended paper in case1.

In case 2,  It's my mistake, I ignored the segs from deep-photo-style-transfer. And the result looks fine now.

in case 3, the bottom row is exactly stylized by the yellow bed, but the top row looks more photorealistic. 
Thx.
